### PR TITLE
Ui outline + redux file reorg

### PR DESCRIPTION
**Pull request by Schleppguy**
@kylev sorry this PR is so beefy.

### CODE REVIEW

File: src/exampleTracks.js [hunk 1]
```diff
@@ -0,0 +1,2878 @@
+export const exampleTracks = [
```
Comment (kylev) [inline src/exampleTracks.js]: Node can import JSON. Just put this JSON in a `exampleTracks.json` and `import exampleTracks from "./exampletTracks.json"`.

File: src/actions.js [hunk 2]
```diff
@@ -1,3 +1,55 @@
 import { createAction } from 'redux-actions';
+import SC from 'soundcloud';
+import _ from 'lodash';
 
 export const addFun = createAction('ADD_FUN');
+export const getTracksFromSC = createAction('GET_TRACKS_FROM_SC');
```
Comment (kylev) [inline src/actions.js]: Why does this file now contain all your SC library code? `actions.js` should only contain redux action creator methods.

File: src/containers/Player.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (kylev) [inline src/containers/Player.js]: Write this as a dumb component first.

In general, JSX in `components/` and just JS in `containers/`. You're going to need to able to test the player's behavior via props, and you can't do that if it isn't in a separate component from the redux-connected container.

File: src/containers/SCButton.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Comment (kylev) [inline src/containers/SCButton.js]: Again, you're mixing component (markup) with container (data flow).

Perhaps renaming this code to `component/SCButton` and `container/TrackLoader` will help. Then write a new component called `component/SimpleButton`. It should be EASY to swap between two "dumb display things that call a passed `getTracks` method" underneath the redux container `TrackLoader`.

You should be able to swap `SCButton` and `SimpleButton` out with an `import` change and a different argument to HOC creator `connect()` thingy.

File: src/actions.js [hunk 5]
```diff
@@ -1,3 +1,55 @@
...
+// async
+const DEFAULT_USER_ID = '7742327';
+
+SC.initialize({
+  client_id: process.env.REACT_APP_SC_CLIENT_ID,
+  redirect_uri: process.env.REACT_APP_REDIRECT_URI
+});
+
+export const getTracks = () => {
+  return dispatch => {
```
Comment (kylev) [inline src/actions.js]: `getTracks()` is no longer testable on its own. You now need a whole redux dispatcher system set up to run that test.

**Review by kylev — APPROVED** [review]
It'll work, but you moved backwards on code isolation. Leave the SC code alone and work on the promises it returns. You ended up back in the place where you're typing near-identical code in two branches of an if-statement.

Not ready to merge.

**Review by kylev — CHANGES_REQUESTED** [review]
Meant to request changes...

File: src/actions.js (hunk 5 above)
Comment (kylev) [reply src/actions.js]: I think your goal should be a thunk action like this:

```js
export const getTracks = () {
  return dispatch => {
    scLibraryMethod
      .then(tracks => dispatch(getTracksFromSC(tracks))
      .catch(e => dispatch(getTracksFromSC(e));    	 
  }
}
```

File: src/actions.js (hunk 2 above)
Comment (kylev) [reply src/actions.js]: Also, consider renaming this action to something like "APPEND_NEW_TRACKS". The "getting" has been taken over by the thunk action. This action (and the reducer that handles it) just do work on the state's track list and a payload of tracks.

File: src/containers/TrackList.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { connect } from 'react-redux';
```
Comment (kylev) [inline src/containers/TrackList.js:1]: This is gorgeous. Just the wiring up of the component, exactly as we discussed!

**Review by kylev — APPROVED** [review]
Nice. Looks like you got everything. Ship it!

File: src/containers/TrackList.js (hunk 6 above)
Comment (Schleppguy) [reply src/containers/TrackList.js:1]: Yay!

**Pull request merged**